agent/action/mount_disk.go

```diff
@@ -59,11 +64,23 @@ func (a MountDiskAction) Run(diskCid string) (interface{}, error) {
 
 	mountPoint := a.dirProvider.StoreDir()
 
-	isMountPoint, err := a.mountPoints.IsMountPoint(mountPoint)
+	realPath, _, err := a.devicePathResolver.GetRealDevicePath(diskSettings)
+	if err != nil {
+		return nil, bosherr.WrapError(err, "Getting real device path")
+	}
+
+	devicePath, isMountPoint, err := a.diskMounter.IsMountPoint(mountPoint)
 	if err != nil {
 		return nil, bosherr.WrapError(err, "Checking mount point")
 	}
+	a.logger.Info(logTag, "realPath = %s, devicePath = %s, isMountPoint = %s", realPath, devicePath, isMountPoint)
+
 	if isMountPoint {
+		if realPath == devicePath || fmt.Sprint(realPath+"1") == devicePath {
```

Comment:
@kaleo211 what is the purpose of comparing devicePath with realPath+1? It is unclear from code and there are no tests that are failing when removing this condition.